### PR TITLE
Patch avahi for multiple CVEs

### PR DESCRIPTION
**Pull request by Kanishk-Bansal**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- CVE-2023-38469
- CVE-2023-38470
- CVE-2023-38471
- CVE-2023-38472
- CVE-2023-38473

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-38469
- https://nvd.nist.gov/vuln/detail/CVE-2023-38470
- https://nvd.nist.gov/vuln/detail/CVE-2023-38471
- https://nvd.nist.gov/vuln/detail/CVE-2023-38472
- https://nvd.nist.gov/vuln/detail/CVE-2023-38473

###### Test Methodology

- Pipeline build id: [686614](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=686614&view=results)

### CODE REVIEW

File: SPECS/avahi/avahi.spec [hunk 1]
```diff
@@ -420,6 +425,21 @@ exit 0
...
+
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> -0.8-4
+- Fix CVE-2023-38473 with an upstrem patch
+
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> -0.8-4
+- Fix CVE-2023-38472 with an upstream patch
+
+* Wed Nov 27 2024 Kanishk Bansal <kanbansal@microsoft.com> - 0.8-4
+- Fix CVE-2023-38469 with an upstream patch
+
```
Comment (mbykhovtsev-ms) [inline SPECS/avahi/avahi.spec:442]: Usually we list multiple changes as single changelog entry within same release using the - as list entries like this
```suggestion
* Wed Nov 27 2024 Kanishk Bansal <kanbansal@microsoft.com> - 0.8-4
- Fix CVE-2023-38469 with an upstream patch
- Fix CVE-2023-38470 with an upstream patch
- Fix CVE-2023-38471 with an upstream patch
- Fix CVE-2023-38472 with an upstream patch
- Fix CVE-2023-38473 with an upstream patch
```

**Review by mbykhovtsev-ms — CHANGES_REQUESTED** [review]

File: SPECS/avahi/avahi.spec (hunk 1 above)
Comment (Kanishk-Bansal) [reply SPECS/avahi/avahi.spec]: done

File: SPECS/avahi/avahi.spec [hunk 2]
```diff
@@ -420,6 +425,13 @@ exit 0
 %endif
 
 %changelog
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> -0.8-4
+- Fix CVE-2023-38471 wih an upstream patch
+- Fix CVE-2023-38470 with an upstream patch
+- Fix CVE-2023-38473 with an upstrem patch
+- Fix CVE-2023-38472 with an upstream patch
+- Fix CVE-2023-38469 with an upstream patch
```
Comment (mbykhovtsev-ms) [inline SPECS/avahi/avahi.spec:433]: Typos
```suggestion
* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> - 0.8-4
- Fix CVE-2023-38471 with an upstream patch
- Fix CVE-2023-38470 with an upstream patch
- Fix CVE-2023-38473 with an upstream patch
- Fix CVE-2023-38472 with an upstream patch
- Fix CVE-2023-38469 with an upstream patch
```

Comment (Kanishk-Bansal) [reply SPECS/avahi/avahi.spec]: Addressed!

File: SPECS/avahi/avahi.spec [hunk 3]
```diff
@@ -420,6 +425,13 @@ exit 0
 %endif
 
 %changelog
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> -0.8-4
+- Fix CVE-2023-38471 wtih an upstream patch
```
Comment (mbykhovtsev-ms) [inline SPECS/avahi/avahi.spec:429]: Couple more typos. 
```suggestion
* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> - 0.8-4
- Fix CVE-2023-38471 with an upstream patch
```

Comment (Kanishk-Bansal) [reply SPECS/avahi/avahi.spec]: Done

File: SPECS/avahi/avahi.spec [hunk 4]
```diff
@@ -420,6 +425,13 @@ exit 0
 %endif
 
 %changelog
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> -0.8-4
```
Comment (mbykhovtsev-ms) [inline SPECS/avahi/avahi.spec:428]: Last typo
```suggestion
* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> - 0.8-4
```

Comment (Kanishk-Bansal) [reply SPECS/avahi/avahi.spec]: Done

**Review by mbykhovtsev-ms — APPROVED** [review]

File: SPECS/avahi/avahi.spec [hunk 5]
```diff
@@ -420,6 +425,13 @@ exit 0
 %endif
 
 %changelog
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> - 0.8-4
+- Fix CVE-2023-38471 with an upstream patch
```
Comment (jslobodzian) [inline SPECS/avahi/avahi.spec:429]: Please combine these into a single line in the future.

**Pull request merged**